query-connector/src/app/queryBuilding/components/SelectionTable.tsx

```diff
@@ -0,0 +1,188 @@
+        return vs;
+      });
+    } else {
+      groupedValueSet.items = Object.values(updatedVS.items).map((vs) => {
+        batchToggleConcepts(vs);
+        vs.includeValueSet = !vs.includeValueSet;
+        return vs;
+      });
+    }
+
+    groupedValueSetsForCondition[valueSetType] = {
+      ...groupedValueSetsForCondition[valueSetType],
+      [key]: groupedValueSet,
+    };
+
+    setValueSets((prevState) => {
+      return {
+        ...prevState,
+        [conditionId]: {
+          ...prevState?.[conditionId],
+          [valueSetType]: {
+            ...prevState?.[conditionId]?.[valueSetType],
+            [key]: updatedVS,
+          },
+        },
+      };
+    });
+  };
+
+  const generateAccordionItems = (types: Array<DibbsValueSetType>) => {
+    const typesWithContent =
+      types &&
+      types.filter(
+        (type) => Object.values(groupedValueSetsForCondition[type]).length > 0,
+      );
+
+    const ValueSetAccordionItems =
+      typesWithContent &&
+      typesWithContent.map((valueSetType) => {
+        const valueSetsForType = Object.values(
+          groupedValueSetsForCondition[valueSetType],
+        );
+        const totalCount = tallyConcpetsForValueSetGroup(
+          valueSetsForType,
+          false,
+        );
+        const selectedCount = tallyConcpetsForValueSetGroup(
+          valueSetsForType,
+          true,
+        );
+
+        const title = (
+          <SelectionViewAccordionHeader
+            valueSetType={valueSetType}
+            conditionId={conditionId}
+            totalCount={totalCount}
+            selectedCount={selectedCount}
+            valueSetsForType={valueSetsForType}
+            handleCheckboxToggle={handleGroupCheckboxToggle}
+            expanded={expanded?.indexOf(valueSetType) > -1 || false}
+          />
+        );
+        const content = (
+          <SelectionViewAccordionBody
+            valueSetType={valueSetType}
+            handleCheckboxToggle={handleSingleCheckboxToggle}
+            valueSetsForType={valueSetsForType}
+          />
+        );
+        const level: HeadingLevel = "h4";
+
+        const handleToggle = (e: React.MouseEvent) => {
+          const element = e.currentTarget.getAttribute("data-testid");
+          const startIndex = element?.indexOf(valueSetType) || 0;
+          const endIndex = valueSetType.length + startIndex;
+
+          if (expanded == valueSetType) {
+            // if the group we clicked on is currently expanded,
+            // toggle it closed
+            setExpandedGroup("");
+          } else {
+            // otherwise, expand the thing we clicked on
+            setExpandedGroup(element?.slice(startIndex, endIndex) || "");
+          }
+        };
+
+        return {
+          title,
+          content,
+          expanded: false,
+          id: `${valueSetType}-${conditionId}`,
+          headingLevel: level,
+          handleToggle,
+        };
+      });
+    return ValueSetAccordionItems;
```

Comment:
`ValueSetAccordionItems` is an array of the objects returned from line 159; elsewhere in the code we render an Accordion component for each item in a map; so each Accordion's `items` property is an array of a single object.

By giving **one** Accordion an array of objects, we can take advantage of built-in features like the multi-select being disabled. (See [here](https://github.com/trussworks/react-uswds/blob/931d2456d1b97c73a54aa598f8bfd7ba1fa50531/src/components/Accordion/Accordion.tsx#L83) for source code)